pulsar/internal/connection.go

```diff
@@ -523,7 +523,7 @@ func (c *connection) internalReceivedCommand(cmd *pb.BaseCommand, headersAndPayl
 		c.handleResponse(cmd.Success.GetRequestId(), cmd)
 
 	case pb.BaseCommand_PRODUCER_SUCCESS:
-		if !*cmd.ProducerSuccess.ProducerReady {
+		if !cmd.ProducerSuccess.GetProducerReady() {
 			request, ok := c.findPendingRequest(cmd.ProducerSuccess.GetRequestId())
 			if ok {
```

Comment:
When `cmd.ProducerSuccess.ProducerReady` is `nil`, it means the response is from broker < 2.8.0, in this case, `handleResponse` is not called, so the pending `Produce` request will never be deleted.

We should differ the case when `ProducerReady` is `nil` and `ProducerReady` is `false`:
- `nil`: call `handleResponse`
- `false`: trigger the callback directly

```go
        case pb.BaseCommand_PRODUCER_SUCCESS:
                resp := cmd.ProducerSuccess
                if resp.ProducerReady == nil || *resp.ProducerReady {
                        c.handleResponse(cmd.ProducerSuccess.GetRequestId(), cmd)
                } else {
                        request, ok := c.findPendingRequest(cmd.ProducerSuccess.GetRequestId())
                        if ok {
                                request.callback(cmd, nil)
                        }
                }
```

/cc @shibd